gordon/metrics/log.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+
+import zope.interface
+
+from gordon import interfaces
+
+# TODO: update plugin loader to load metrics logger as default
+
+
+@zope.interface.implementer(interfaces.ITimer)
+class LogTimer:
+    """Timer that outputs metrics to the logger.
+
+    Args:
+        metric (dict): Metric to output.
+        logger (.MetricLogger): Object used for outputting metrics.
+        time_unit (number): (optional) Multiplier used to scale time.time()
+            output.
+    """
+    def __init__(self, metric, logger, time_unit=1):
+        self.metric = metric
+        self.logger = logger
+        self.time_unit = time_unit
+
+    async def __aenter__(self):
+        await self.start()
+        return self
+
+    async def __aexit__(self, *args):
+        await self.stop()
+
+    async def start(self):
+        """Start the timer.
+        """
+        self._start_time = time.time()
+
+    async def stop(self):
+        """Stop the timer and output the time delta to the logger.
+        """
+        time_elapsed = (time.time() - self._start_time) * self.time_unit
+        self.metric['value'] = time_elapsed
+        self.logger.log(self.metric)
+
+
+class MetricLogger:
+    """Configurable wrapper that formats and outputs metrics.
+
+    Args:
+        level (str): Log level at which to output metric.
+        precision (str): Used to round metric to desired number of
+            places.
+    """
+    LOGFMT = '[{metric_name}]\tvalue: {value:.<prec>f}\tattrs: {attrs}'
+
+    def __init__(self, level, precision):
+        self._log_fmt = self.LOGFMT.replace('<prec>', str(precision))
+        self._log_fn = getattr(logging, level.lower())
+
+    def log(self, metric):
+        """Format and output metric.
+
+        Args:
+            metric (dict): Complete metric.
+        """
+        message = self._log_fmt.format(**metric)
+        self._log_fn(message)
+
+
+@zope.interface.implementer(interfaces.IMetricRelay)
+class LogRelay:
+    """Manager metrics that get output to the logger.
+
+    Args:
+        config (dict): Required keys:
+            level (str): Log level of metrics log messages.
+            precision (int): Number of decimal places to round metric
+                value to for output.
+            time_unit (number): Scale unit to use with timers.
```

Comment:
Could you add to the example config and update prose docs?

Reply:
Also `time_unit` description is a little confusion, could it be similar/same as [this here](https://github.com/spotify/gordon/pull/22/commits/4e78b9e15c1870d65d56b7a12c260ceb722e4f87#diff-6c97c87bd0fada6021501e9bc1ba0925R73)?

Reply:
Hey @econchick, I think a rebase nuked that link. Did you mean something like `Scale time unit for use with time.time() (example: 1E9 to send nanoseconds)`?

Reply:
> Could you add to the example config and update prose docs?

Will do! As soon as #22 is merged and I know what's the preferred way to do this.

Reply:
Here too w/r/t `ns`, `ms`, and `s`.